File: connectx-backend/src/main/java/com/connectx/device/dto/UserPublicKeyDto.java
package com.connectx.device.dto;

public class UserPublicKeyDto {

    private Long deviceId;
    private Long userId;
    private String deviceName;
    private String publicKey;
    private String keyAlgorithm;

    public UserPublicKeyDto() {}

    public UserPublicKeyDto(Long deviceId, Long userId, String deviceName, String publicKey, String keyAlgorithm) {
        this.deviceId = deviceId;
        this.userId = userId;
        this.deviceName = deviceName;
        this.publicKey = publicKey;
        this.keyAlgorithm = keyAlgorithm;
    }

    public Long getDeviceId() {
        return deviceId;
    }

    public void setDeviceId(Long deviceId) {
        this.deviceId = deviceId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getDeviceName() {
        return deviceName;
    }

    public void setDeviceName(String deviceName) {
        this.deviceName = deviceName;
    }

    public String getPublicKey() {
        return publicKey;
    }

    public void setPublicKey(String publicKey) {
        this.publicKey = publicKey;
    }

    public String getKeyAlgorithm() {
        return keyAlgorithm;
    }

    public void setKeyAlgorithm(String keyAlgorithm) {
        this.keyAlgorithm = keyAlgorithm;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/device/model/UserPublicKeyDto.kt
package com.connectx.app.data.remote.device.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.device.dto.UserPublicKeyDto exactly. Distinct from
 * data.remote.user.model.UserIdentityKeyDto -- that DTO is the user's single
 * account-level master key pair (from UserController), while this one is a
 * per-DEVICE public key entry (one row per registered device, from
 * DeviceController#getUserPublicKeys). Genuinely different shape and semantics,
 * so no reuse was applicable.
 */
@Serializable
data class UserPublicKeyDto(
    val deviceId: Long,
    val userId: Long,
    val deviceName: String? = null,
    val publicKey: String? = null,
    val keyAlgorithm: String? = null
)
